sdks/java/io/snowflake/src/test/java/org/apache/beam/sdk/io/snowflake/test/StreamingSnowflakeIOIT.java

```diff
@@ -0,0 +1,195 @@
+  private static String storageIntegrationName;
+
+  @BeforeClass
+  public static void setupAll() throws SQLException {
+    PipelineOptionsFactory.register(SnowflakeIOITPipelineOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(SnowflakeIOITPipelineOptions.class);
+
+    dc =
+        SnowflakeIO.DataSourceConfiguration.create(SnowflakeCredentialsFactory.of(options))
+            .withServerName(options.getServerName())
+            .withDatabase(options.getDatabase())
+            .withRole(options.getRole())
+            .withWarehouse(options.getWarehouse())
+            .withSchema(options.getSchema());
+
+    stagingBucketName = options.getStagingBucketName();
+    storageIntegrationName = options.getStorageIntegrationName();
+
+    for (int i = 0; i < options.getNumberOfRecords(); i++) {
+      testRows.add(TestRow.create(i, String.format("TestRow%s:%s", i, UUID.randomUUID())));
+    }
+
+    TestUtils.runConnectionWithStatement(
+        dc.buildDatasource(),
+        String.format("CREATE OR REPLACE TABLE %s(id INTEGER, name STRING)", TABLE));
+  }
+
+  @AfterClass
+  public static void cleanUp() throws Exception {
+    String bucketNameAndPath = stagingBucketName.replaceAll(".+//", "");
+    String[] parts = bucketNameAndPath.split("/", -1);
+    String bucketName = parts[0];
+    String directory = null;
+    if (parts.length > 1) {
+      directory = bucketNameAndPath.replace(bucketName + "/", "");
+    }
+    TestUtils.clearStagingBucket(bucketName, directory);
+
+    TestUtils.runConnectionWithStatement(
+        dc.buildDatasource(), String.format("DROP TABLE %s", TABLE));
+  }
+
+  @Test
+  public void writeStreamThenRead() throws SQLException, InterruptedException {
+    writeStreamToSnowflake();
+    readAndVerify();
+  }
+
+  private void writeStreamToSnowflake() {
+    TestStream<TestRow> stringsStream =
+        TestStream.create(SerializableCoder.of(TestRow.class))
+            .advanceWatermarkTo(Instant.now())
+            .addElements(
+                testRows.get(0), testRows.subList(1, testRows.size()).toArray(new TestRow[0]))
+            .advanceWatermarkToInfinity();
+
+    pipeline
+        .apply(stringsStream)
+        .apply(
+            "Write SnowflakeIO",
+            SnowflakeIO.<TestRow>write()
+                .withDataSourceConfiguration(dc)
+                .withUserDataMapper(getTestRowDataMapper())
+                .withSnowPipe(options.getSnowPipe())
+                .withStorageIntegrationName(storageIntegrationName)
+                .withStagingBucketName(stagingBucketName)
+                .withFlushTimeLimit(Duration.millis(18000))
+                .withFlushRowLimit(50000)
+                .withDebugMode(StreamingLogLevel.ERROR));
+
+    PipelineResult pipelineResult = pipeline.run(options);
+    pipelineResult.waitUntilFinish();
+  }
+
+  private void readAndVerify() throws SQLException, InterruptedException {
+    int timeout = TIMEOUT;
+    while (timeout > 0) {
```

Comment:
It could be replaced with `for (int timeout = TIMEOUT; timeout > 0; timeout -= INTERVAL)`

Reply:
I think it can be as it is